### DISCUSSION TITLE
Sampling from normal distribution

### DISCUSSION THREAD

**Discussion by mikhailsirenko** [quaquel/EMAworkbench#341] [Q&A] [answered]
Hi team,

Imagine the following setup. My model has a float parameter, and I have a prior that it forms a normal distribution with loc=0 and scale=1. Thus, I would like to sample it in my experiments. What are my options to sample it? Shall I specify it at the level of the RealParameter class? I see there the following:

`self.dist = sp.stats.uniform(lower_bound, upper_bound - lower_bound)  # @UndefinedVariable`

However, I don't see an option to pass an arbitrary distribution yet. Also, the names `lower_bound` and `upper_bound` are not perfect even if there would be an option to pass, e.g., `dist = sp.stats.norm`.

In addition, how should this (`dist` of `RealParameter`) interact with the Sampler class? The `sample` function of, e.g., `LHSSampler`, does accept the `distribution` argument. However, I believe that `distribution` here is something else. Also, when `LHSSampler` does not accept arguments and `perform_experiments` throws an error when passing `LHSSampler().sample(stats.norm(loc=0, scale=1), 1000)`

Did I overlook something, or is there an example of doing this?

(Attaching an example code just in case)

```
from scipy import stats
from ema_workbench.em_framework.samplers import LHSSampler, AbstractSampler, UniformLHSSampler, MonteCarloSampler
from ema_workbench import Model, ScalarOutcome, RealParameter, perform_experiments
import matplotlib.pyplot as plt

# Define a simple model
def simple_model(x, y):
    return {'result': x + y}


# Create an EMA model
my_model = Model(name="SimpleModel", function=simple_model)

# Define the uncertainties
my_model.uncertainties = [RealParameter("x", lower_bound=0, upper_bound=10),
                          RealParameter("y", lower_bound=100, upper_bound=200)]

# Define the outcomes
my_model.outcomes = [ScalarOutcome("result")]

# Define the sampler
sampler = LHSSampler().sample(stats.norm(loc=0, scale=1), 1000)

# Perform experiments
results = perform_experiments(my_model, 1000, uncertainty_sampling=sampler)
```

**Comment by quaquel** [comment, marked as answer]
You can use `RealParameter.from_dist`. See [Eigenraam example](https://github.com/quaquel/EMAworkbench/blob/master/ema_workbench/examples/example_eijgenraam.py)

**Reply by mikhailsirenko** [reply]
Awesome! Solved!
